instrumentation/rocketmq-client-4.8/library/src/main/java/io/opentelemetry/instrumentation/rocketmq/TracingConsumeMessageHookImpl.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.rocketmq;
+
+import static io.opentelemetry.instrumentation.rocketmq.RocketMqConsumerTracer.tracer;
+
+import io.opentelemetry.context.Context;
+import org.apache.rocketmq.client.hook.ConsumeMessageContext;
+import org.apache.rocketmq.client.hook.ConsumeMessageHook;
+
+public class TracingConsumeMessageHookImpl implements ConsumeMessageHook {
+
+  @Override
+  public String hookName() {
+    return "OpenTelemetryConsumeMessageTraceHook";
+  }
+
+  @Override
+  public void consumeMessageBefore(ConsumeMessageContext context) {}
+
+  @Override
+  public void consumeMessageAfter(ConsumeMessageContext context) {
+    if (context == null || context.getMsgList() == null || context.getMsgList().isEmpty()) {
+      return;
+    }
+    Context traceContext = tracer().startSpan(Context.current(), context.getMsgList());
+    tracer().end(traceContext);
```

Comment:
The actual processing logic will not be included in the trace if you start and immediately close the span. How about using `ConsumeMessageContext#mqTraceContext` field to store the context and scope?